src/attrib.d

```diff
@@ -1480,6 +1480,25 @@ extern (C++) static uint setMangleOverride(Dsymbol s, char* sym)
         s.isDeclaration().mangleOverride = sym;
         return 1;
     }
+    else if (s.isStructDeclaration() || s.isClassDeclaration())
+    {
+        bool asClass = !strcmp(sym, "class");
+        bool asStruct = !strcmp(sym, "struct");
+        if (!(asClass || asStruct))
+            return 0;
+        if (auto sd = s.isStructDeclaration())
+            sd.cppmangleAsClass = asClass;
+        else if (auto cd = s.isClassDeclaration())
+            cd.cppmangleAsStruct = asStruct;
+        return 1;
+    }
+    else if (s.isTemplateDeclaration() && s.isTemplateDeclaration().onemember)
+    {
+        Dsymbol ds = s.isTemplateDeclaration().onemember;
+        if (ds.isStructDeclaration() || ds.isClassDeclaration())
+            return setMangleOverride(ds, sym);
+        return 0;
```

Comment:
This branch is a completely new behavior, and the root of the hack in `syntaxCopy`.

Currently `pragma(mangle)` does not affect beyond the template definition boundary.
```d
pragma(mangle, "V1") int var;
pragma(mangle, "V2") template V2() { int var; }
template V3() { pragma(mangle, "V3") int var; }

pragma(mangle, "F1") void func() {}
pragma(mangle, "F2") template F2() { void func() {} }
template F3() { pragma(mangle, "F3") void func() {} }

pragma(msg, var.mangleof);          // V1
pragma(msg, V2!().var.mangleof);    // pragma(mangle) has no effect
pragma(msg, V3!().var.mangleof);    // V3

pragma(msg, func.mangleof);         // V1
pragma(msg, F2!().func.mangleof);   // pragma(mangle) has no effect
pragma(msg, F3!().func.mangleof);   // V3
```

But this branch adds a new special behavior only for classes and structs.
If we follow existing `pragma(mangle)` beahvior, this branch should be removed.